freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java

```diff
@@ -0,0 +1,167 @@
+package com.sequenceiq.freeipa.service.image;
+
+import com.sequenceiq.freeipa.api.model.image.FreeIpaVersions;
+import com.sequenceiq.freeipa.api.model.image.Image;
+import com.sequenceiq.freeipa.api.model.image.ImageCatalog;
+import com.sequenceiq.freeipa.api.v1.freeipa.stack.model.common.image.ImageSettingsRequest;
+import com.sequenceiq.freeipa.dto.ImageWrapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.inject.Inject;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class FreeIpaImageProvider implements ImageProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FreeIpaImageProvider.class);
+
+    private static final Pattern VERSION_PATTERN = Pattern.compile("^([0-9]+\\.[0-9]+\\.[0-9]+-(dev\\.|rc\\.|[b]))[0-9]+$");
+
+    @Inject
+    private ImageCatalogProvider imageCatalogProvider;
+
+    @Value("${freeipa.image.catalog.url}")
+    private String defaultCatalogUrl;
+
+    @Value("${freeipa.image.catalog.default.os}")
+    private String defaultOs;
+
+    @Value("${info.app.version:}")
+    private String freeIpaVersion;
+
+    @Override
+    public Optional<ImageWrapper> getImage(ImageSettingsRequest imageSettings, String region, String platform) {
+        String imageId = imageSettings.getId();
+        String catalogUrl = Objects.nonNull(imageSettings.getCatalog()) ? imageSettings.getCatalog() : defaultCatalogUrl;
```

Comment:
nit: Objects.requireNonNullElse() would be better

Reply:
Copy-paste from ImageService